lib/collection/src/shards/replica_set/update.rs

```diff
@@ -580,6 +580,14 @@ impl ShardReplicaSet {
                 continue;
             }
 
+            // Ignore missing point errors if replica is in partial or recovery state
+            // Partial or recovery state indicates that the replica is receiving a shard transfer,
+            // it might not have received all the points yet
+            // See: <https://github.com/qdrant/qdrant/pull/5991>
+            if peer_state.is_partial_or_recovery() && err.is_missing_point() {
+                continue;
+            }
```

Comment:
The change is simple: do not mark a replica as dead on a missing point error if it's in partial/recovery state.